one-ui/ui/src/components/entities/mapping/create-edit-mapping-dialog/create-edit-mapping-dialog.tsx

```diff
@@ -320,8 +321,8 @@ const CreateEditMappingDialog = (props) => {
           Name:&nbsp;<span className={styles.asterisk}>*</span>
           &nbsp;
             </span>} labelAlign="left"
-          validateStatus={(mapName || !isMapNameTouched) ? '' : 'error'}
-          help={(mapName || !isMapNameTouched) ? '' : 'Name is required'}
+          validateStatus={(mapName || !isMapNameTouched) ? (props.title === 'Edit Mapping' ? '' : (!props.mapNames.hasOwnProperty(mapName) ? '' : 'error')) : 'error'}
+          help={(mapName || !isMapNameTouched) ? (props.title !== 'Edit Mapping'? (props.mapNames.hasOwnProperty(mapName) ? 'A mapping step with the same name already exists. Enter a unique name.':'') : '') : 'Name is required'}
```

Comment:
Is this doing its own form of validation? The server is validating this already, and throwing an error with a message of "A mapping with the same name but for a different entity type already exists. Please choose a different name.". 

If this is doing the validation by comparing the mapping names to the known set of names already - I don't think that's 100% reliable, because a new mapping could have been created between the time that the UI fetches the set of names and the user then tries to create one. That's why the server has to do the validation. The UI should just catch the error from the server and then display it.

Reply:
+1

Reply:
@bsrikan - The way it works currently is that as soon as you create a mapping artifact, the react refreshes the entity page showing the new map artifact created. By this time the mapping names object is already refreshed along with the entire mapping list for that entity or else you will not even see the newly created artifact in UI.
This approach is really good for validating on the fly as user types the names. Otherwise use will have to click on Save button several times to hit the right name.

However, we can add another validation on top of this on Save button, just to take care of the case where the newly added artifact has not shown up yet in the UI and for some reason user again tries to add another artifact with same name.
Let me know your thoughts.

Reply:
One problem is that multiple users could try to create the same mapping at the same time. The UI shouldn't do any validation here. It should depend on the server to ensure that two mappings with the same name can't be created. The server will throw an error when that happens, and the UI should then display the error message from the server.

Reply:
Just handling the error based on the backend server validation (as Rob describes) seems like the cleaner way to do this. I don't think this error is common enough to require the on-the-fly checking. @xnikhil08 are you OK with that?

Reply:
@rjrudin - OK, but the current server validation throws error only when the same mapping name exists in another entity but doesn't throw any error when it's the same entity. I am assuming we need this validation for same entity as well right?

Reply:
@wooldridge - Yeah i am fine with it. I guess my only concern was the user experience, if there are a lot more mappings then user would have to keep clicking on the save button to get to the unique value. Thats why now i was thinking of having a sort of combination of both backend and on-the-fly validation.
But yeah we can remove the on-the-fly validation completely if thats not needed here.

Reply:
@xnikhil08 if the backend can be fixed to recognize the case where there are same-name mappings for the same entity, then I think we should just let the backend handle the validation and keep the UI code cleaner.

Reply:
@rjrudin & @bsrikan & @wooldridge - Please review now. Updated the PR as discussed.